### ISSUE TITLE
Ikhal doesn't save changes in start or end times if you exit the widget with Page Down

### ISSUE THREAD

**Issue by Pi2048** [pimutils/khal#1345]
**Describe the bug**
If I open an event for editing in ikhal, and I change the start or end time, this change is sometimes not saved. This occurs if I exit the widget for the start or end time by using Page Down instead of the arrow keys.

**If applicable: Stack Trace**
n/a

**To Reproduce**

These are the steps for the From: time, but the bug also applies to the To: time.

* Start ikhal
* Move the cursor to an existing event with a start and end time (i.e. not an all-day event) and press enter to view details.
* Press enter again to open the event editor
* Use the arrow keys to navigate to the From: time field.
* Change its value slightly, e.g. change 18:10 to 18:12
* Navigate away from the field, but use Page Down instead of the arrow keys. The cursor ends up on the Export button.
* Press the up arrow key once to land on the Save button.
* Press enter to press the Save button.

**Expected behavior**
I expected the change I made to the From: time to be saved and displayed in the calendar overview. Instead, the change I made is silently dropped.

**Screenshots**

**OS, version, khal version and how you installed it:**
 - The output of khal --version: `khal, version 0.11.3`
 - Installation method: PyPi
 - python version: Python 3.10
 - OS: Ubuntu 22.04
 - Your khal config file: I use a minimal config file with two calendars. The only fields that are relevant to this bug are:
 ```
[locale]

timeformat="%H:%M"
```
 - The versions of your other python packages [e.g. the output of `pip freeze`]
<details>
<summary>Output of `pip freeze`</summary>
androguard==3.4.0a1
appdirs==1.4.4
apsw==3.36.0.post1
apturl==0.5.2
arandr==0.1.10
archive-hocr-tools==1.1.50
archive-pdf-tools==1.5.4
argcomplete==1.8.1
asn1crypto==1.4.0
atomicwrites==1.4.0
attrs==21.2.0
autobahn==21.11.1
Automat==20.2.0
backcall==0.2.0
base58==1.0.3
bcrypt==3.2.0
beautifulsoup4==4.10.0
beniget==0.4.1
binwalk==2.3.3
blinker==1.4
Brlapi==0.8.3
Brotli==1.0.9
cbor==1.0.0
certifi==2020.6.20
cffi==1.15.0
chardet==4.0.0
chrome-gnome-shell==0.0.0
click==8.0.3
click-log==0.2.1
click-repl==0.3.0
click-threading==0.5.0
colorama==0.4.4
command-not-found==0.3
configobj==5.0.6
constantly==15.1.0
cryptography==3.4.8
css-parser==1.0.7
cssselect==1.1.0
cssutils==1.0.2
cupshelpers==1.0
cycler==0.11.0
Cython==0.29.28
dbus-python==1.2.18
decorator==4.4.2
defer==1.0.6
defusedxml==0.7.1
diffoscope==205
distro==1.7.0
distro-info==1.1+ubuntu0.2
dnspython==2.1.0
duplicity==0.8.21
ecdsa==0.18.0b1
EditorConfig==0.12.2
enjarify==1.0.3
fasteners==0.14.1
feedparser==6.0.8
flatbuffers===1.12.1-git20200711.33e2d80-dfsg1-0.6
fonttools==4.29.1
fs==2.4.12
future==0.18.2
gast==0.5.2
GeoIP==1.3.2
gitdb==4.0.9
GitPython==3.1.24
gpg==1.16.0
h2==4.1.0
hkdf==0.0.3
hpack==4.0.0
html2text==2020.1.16
html5-parser==0.4.10
html5lib==1.1
httplib2==0.20.2
humanize==0.0.0
hyperframe==6.0.0
hyperlink==21.0.0
icalendar==4.0.3
idna==3.3
ifaddr==0.1.7
imageio==2.31.1
imaplib2==3.5
img2pdf==0.4.2
importlib-metadata==4.6.4
incremental==21.3.0
iotop==0.6
ipython==7.31.1
itemadapter==0.4.0
itemloaders==1.0.4
jedi==0.18.0
jeepney==0.7.1
jmespath==0.10.0
jsbeautifier==1.14.0
jsondiff==1.3.1
keyring==23.5.0
khal==0.11.3
khard==0.17.0
kiwisolver==1.3.2
language-selector==0.1
launchpadlib==1.10.16
lazr.restfulclient==0.14.4
lazr.uri==1.0.6
lazy_loader==0.3
libarchive-c==2.9
lockfile==0.12.2
louis==3.20.0
lxml==4.8.0
lz4==3.1.3+dfsg
macaroonbakery==1.3.1
magic-wormhole==0.12.0
Mako==1.1.3
Markdown==3.3.6
markdownify==0.11.6
MarkupSafe==2.0.1
matplotlib==3.5.1
matplotlib-inline==0.1.3
mechanize==0.4.7
mnemonic==0.19
monotonic==1.6
more-itertools==8.10.0
mpmath==0.0.0
msgpack==1.0.3
netifaces==0.11.0
networkx==3.1
numpy==1.21.5
oauthlib==3.2.0
olefile==0.46
packaging==21.3
paramiko==2.9.3
parsedatetime==2.6
parsel==1.6.0
parso==0.8.1
passlib==1.7.4
pdfminer.six===-VERSION-
pexpect==4.8.0
pickleshare==0.7.5
pikepdf==5.0.1+dfsg
Pillow==9.0.1
ply==3.11
progressbar==2.5
prompt-toolkit==3.0.43
Protego==0.2.1
protobuf==3.12.4
psutil==5.9.0
ptyprocess==0.7.0
py-ubjson==0.16.1
py7zr==0.11.3+dfsg
pyasn1==0.4.8
pyasn1-modules==0.2.1
pycairo==1.20.1
pychm==0.8.6
pycparser==2.21
pycryptodomex==3.11.0
pycups==2.0.1
PyDispatcher==2.0.5
pydot==1.4.2
Pygments==2.11.2
PyGObject==3.42.1
PyHamcrest==2.0.2
PyJWT==2.3.0
pymacaroons==0.13.0
PyMuPDF==1.22.5
PyNaCl==1.5.0
PyOpenGL==3.1.5
pyOpenSSL==21.0.0
pyparsing==2.4.7
PyPDF2==1.26.0
pyperclip==1.8.2
pypng==0.0.20
PyQRCode==1.2.1
PyQt5==5.15.6
PyQt5-sip==12.9.1
pyqtgraph==0.12.4
PyQtWebEngine==5.15.5
pyRFC3339==1.1
PySocks==1.7.1
pysrt==1.0.1
python-apt==2.4.0+ubuntu3
python-dateutil==2.8.1
python-debian==0.1.43+ubuntu1.1
python-magic==0.4.24
python-snappy==0.5.3
pythran==0.10.0
PyTrie==0.4.0
pytz==2022.1
PyWavelets==1.4.1
pyxattr==0.7.2
pyxdg==0.27
PyYAML==5.4.1
queuelib==1.6.2
regex==2021.11.10
reportlab==3.6.8
repoze.lru==0.7
requests==2.25.1
requests-toolbelt==0.9.1
roman==4.1
Routes==2.5.1
rpm==4.17.0
ruamel.yaml==0.17.16
ruamel.yaml.clib==0.2.6
scikit-image==0.21.0
scipy==1.8.0
Scrapy==2.5.1
SecretStorage==3.3.1
service-identity==18.1.0
setproctitle==1.2.2
sgmllib3k==1.0.0
six==1.16.0
smmap==5.0.0
sortedcontainers==2.1.0
soupsieve==2.3.1
spake2==0.8
srt==3.5.3
sympy==1.9
systemd-python==234
tabulate==0.8.9
terminator==2.1.1
texttable==1.6.4
tifffile==2023.7.10
tlsh==0.2.0
todoman==4.4.0
torbrowser-launcher==0.3.3
tqdm==4.57.0
traitlets==5.1.1
twilio==7.12.0
Twisted==22.1.0
txaio==21.2.1
txtorcon==20.0.0
tzlocal==2.1
u-msgpack-python==2.3.0
ubuntu-drivers-common==0.0.0
ubuntu-pro-client==8001
ufoLib2==0.13.1
ufw==0.36.1
ujson==5.1.0
unattended-upgrades==0.1
unicodedata2==14.0.0
Unidecode==1.3.3
urllib3==1.26.5
urwid==2.1.2
usb-creator==0.3.7
vboxapi==1.0
vdirsyncer==0.18.0
vobject==0.9.6.1
vosk==0.3.45
w3lib==1.22.0
wadllib==1.3.6
wcwidth==0.2.5
webencodings==0.5.1
WebOb==1.8.6
websockets==11.0.3
wsaccel==0.6.3
wxPython==4.0.7
xdg==5
xkit==0.0.0
xlrd==1.2.0
xmltodict==0.13.0
zeroconf==0.38.3
zipp==1.0.0
zope.interface==5.4.0
</details>

**Additional context**
Add any other context about the problem here. Especially, if the issue came up when reading an .ics file, please provide the content of that file (anonymize if needed).

**Comment by Pi2048** [comment]
This is possibly a duplicate of #1274

**Comment by geier** [comment]
should be fixed with #1346
